### PR TITLE
To Do list: Add And Remove

### PR DESCRIPTION
**Pull request by HENRYKC24**
### In this pull request, I >
- Removed all hardcoded items from the tasks array.
- Created a new JavaScript file for the new functionality.
- Implemented a function for adding a new task.
- Implemented a function for editing task descriptions.
- Implemented a function for deleting a task.
- Implemented a function for the "Clear all completed" button.
- By default, the new task index is taking the highest index plus one to avoid having the same indices in the system.
- All changes to the To-Do List are be saved in local storage

### CODE REVIEW

File: src/manipulateToDo.js [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+  if (!description) {
+    goToInput();
+    return false;
+  }
+  const indexArray = tasks.map(({ index }) => index);
+  const sortedIndeces = indexArray.sort((a, b) => a - b);
+  let highestIndex = sortedIndeces[sortedIndeces.length - 1];
+  if (Number.isNaN(highestIndex) || !highestIndex) {
+    highestIndex = 0;
+  }
```
Comment (hemantso) [inline src/manipulateToDo.js]: - [ ] After deleting the selected task on clicking `clear all completed`, It works fine to delete them. But when I add a new task the deleted task comes up with a new item. This is strange behaviour with localStorage. Kindly try to fix this issue.

Initial List
image

After deleting the Selected item on clicking `Clear all Completed`
image

After adding again a new task
image

**Review by hemantso — CHANGES_REQUESTED** [review]
Hi @HENRYKC24, 

Good job so far! 👍 
There are some issues that you still need to work on to go to the next project but you are almost there! 

Alt Text
## Required Changes ♻️
_Check the comments under the review._

Cheers and happy coding 😃

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. You can mention me (@hemant-soni-vst-au4) so I can see your comments and respond.


**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/manipulateToDo.js (hunk 1 above)
Comment (HENRYKC24) [reply src/manipulateToDo.js]: Thank you so much, @hemant-soni-vst-au4:pray: 
I have done my best to get the bug fixed. I am heading over to my dashboard to resubmit another request.
I am grateful:raised_hands:

**Review by hemantso — APPROVED** [review]
Hi @HENRYKC24, 

Great work on making the requested changes. 👏 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

## Congratulations! 🎉 

Cheers and happy coding 😃

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. You can mention me (@hemant-soni-vst-au4) so I can see your comments and respond.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by HENRYKC24** [comment]
Thank you and God bless you:pray:

**Pull request merged**